### ISSUE TITLE
Quick Choice FSC: Default value is not displayed when using visual cards

### ISSUE THREAD

**Issue by clifford-fra** [alexed1/LightningFlowComponents#264]
As mentioned in the title:

When a default value is passed to the component, the according visual card does not show as selected (missing the blue border).

The behavior of the component is still as expected. You can just click on next, even if Required is set to {!$GlobalConstant.True}.

**Comment by ericrsmith35** [comment]
I'm not sure how to implement a preselected display with the current component.  Maybe someone could take a look.

**Comment by clifford-fra** [comment]
Hi Eric,

I have almost no experience in javascript and even less in developing lightning web components, but I've spent the last 6 hours trying out a million things... Currently, I'm working on a pretty dirty solution, which someone may pick up and refine.

1. I have added a new CSS Style for the default value:
```
.selected-visual-picker {
    border: 2px solid rgb(21, 137, 238);
    box-shadow: 0 0 0 1px rgb(21, 137, 238) inset; 
}
```

2. In the HTML File, I added a data-id property to the visual cards:
```
<template for:each={items} for:item="item">
    <div key={item.name} class={columnClass} data-id={item.name}>
...
```

3. In the Javascript file, I had to use the renderedCallback() function. I'm checking if showVisual is true and a value is set. If true, I fetch the visual card by its data-id and add the new CSS class from above. This card will be marked as default.
```
renderedCallback() {
    if (this.showVisual && this.value != null) {
        this.template.querySelector('[data-id="' + this.value + '"]').classList.add('selected-visual-picker');
    }
}
```

4. Of course, I have to take care of removing the CSS-Class from the card, as soon as the user selects something different then the default value. So I had to make a small addition to the handleChange(event) function:
```
handleChange(event) {
    if (this.showVisual && this.value != null) {
        this.template.querySelector('[data-id="' + this.value + '"]').classList.remove('selected-visual-picker');
    }
...
```

I have mixed feelings about this solution and the blue border still looks ugly. But hey, it works :-)

I hope I can get a better styling or maybe there's already a class/style I could use?

![first_attempt](https://user-images.githubusercontent.com/55443386/73545977-ae3a3c00-443c-11ea-855e-6f27b69d0cba.png)

**Comment by ericrsmith35** [comment]
That's a great start.  I can't figure out if slds-visual-picker has a setting for showing something as selected.
https://www.lightningdesignsystem.com/components/visual-picker/

**Comment by clifford-fra** [comment]
Well, I believe this is as far as I can get. A small addition:

Instead of changing the HTML file to:

```
<template for:each={items} for:item="item">
    <div key={item.name} class={columnClass} data-id={item.name}>
...
```

It's better to give the `data-id` to the following `<span>`-tags. One for the template for visual cards with icons and one for the template without icons:

```
<!-- Display Visual Card Pickers with Icons-->
<template if:true={includeIcons}>
    <span
        class="slds-visual-picker__figure slds-visual-picker__text class=slds-box slds-box_link slds-box_x-small slds-media"
        style="width:25rem" data-id={item.name} >
...
```
And here:

```
<!-- Display Visual Card Pickers without Icons-->                            
<template if:false={includeIcons}>
    <span class="slds-visual-picker__figure slds-visual-picker__text" data-id={item.name} >
...
```
So finally, if you change the new CSS-Style to 1px border:
```
.selected-visual-picker {
    border: 1px solid rgb(21, 137, 238);
    box-shadow: 0 0 0 1px rgb(21, 137, 238) inset; 
}
```
Everything looks pretty decent. The only thing I can't reverse engineer, is how to display the check icon on the top right of a card:

![second_attempt](https://user-images.githubusercontent.com/55443386/73551936-9ddb8e80-4447-11ea-96f8-05604f956a60.png)

Any help would be nice!

**Comment by ericrsmith35** [comment]
I'll I've been able to find for that is this:
<span class="slds-icon_container slds-visual-picker__text-check"> <svg class
="slds-icon slds-icon-text-check slds-icon_x-small" aria-hidden="true"> <use
xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#check"></use> </svg
> </span>

**Comment by clifford-fra** [comment]
Hi Eric,

I solved the issue and was able to even simplify the code. Actually, there was no need to add or remove new CSS Styles. All, I had to do, was to set the `<input radio>` to `checked = true`. The Lightning Design System will then take care of the styling :-)

Here's the complete fix, so forget what I wrote in the posts before:

1. In the HTML file, we need to add a `data-id` property to the radio-buttons in the Visual Card section, so that we can later on identify these buttons when the component is rendered:

```
...
<template for:each={items} for:item="item">
      <div key={item.name} class={columnClass}>
             <input type="radio" id={item.name} value={item.name} name="visualList" data-id={item.name} onclick={handleChange} />
...
```

2. In the Javascript file, we just need to add the renderedCallback() function. As soon as the component is rendered. The function checks it the mode is set to visual and if a value has been set. If this is true, the according radio button (visual card) will be checked.

```
renderedCallback() {
      if (this.showVisual && this.value != null) {
            this.template.querySelector('[data-id="' + this.value + '"]').checked = true;
      }
}
```
I hope we can add this fix to the package. Would be really nice.

Best regards,
Clifford
